src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -982,12 +990,18 @@ private void AddDependencies(Dictionary<String, Tuple<PackageArchiveReader, Pack
                     }
 
                     var dependency = new PackageDependency(packageReference.PackageIdentity.Id, range);
+                    var packageName = $"{packageReference.PackageIdentity.Id}.{packageReference.PackageIdentity.Version}";
+
+                    var stream = finder.GetNupkgStreamAsync(packageReference.PackageIdentity.Id, packageReference.PackageIdentity.Version, CancellationToken.None).Result;
```

Comment:
Adding using here also causes problems because it closes the stream before it's done reading it.